### DISCUSSION TITLE
Add support for rowspan

### DISCUSSION THREAD

**Discussion by dmrickey** [EddieDover/fvtt-party-sheet#37] [Ideas]
For cases like this where a single actor has multiple rows, it would be ideal if this last column was able to span all three rows so it doesn't force the first row to be "too tall"
![image](https://github.com/user-attachments/assets/a7dffcb9-aee3-467b-b96a-9aeeab6c4139)

edit: Actually, looking at the pic again, it would also be nice to do this for the first column so that it can be vertically centered within its own actor "row"
edit x2: And same for that classes entry, rowspan 2 would be ideal for an actor that has multiple classes

**Comment by EddieDover** [comment]
The reason this hasn't been implemented already is because mechanically it's pretty rough. 

A rowspan only works if there's no row beneath the one you're spanning,
Without:
<table>
<thead>
  <tr>
    <th>A</th>
    <th>B</th>
    <th>Example time</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Lorum</td>
    <td>Ipsum</td>
    <td>I span one</td>
  </tr>
  <tr>
    <td>Lorum</td>
    <td>Ipsum</td>
    <td>But I already exist</td>
  </tr>
</tbody>
</table>
Vs with a span

<table>
<thead>
  <tr>
    <th>A</th>
    <th>B</th>
    <th>Example time</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Lorum</td>
    <td>Ipsum</td>
    <td rowspan="2">I span two</td>
  </tr>
  <tr>
    <td>Lorum</td>
    <td>Ipsum</td>
    <td>But I already exist</td>
  </tr>
</tbody>
</table>

So inside the handlebars template, I'd have to somehow backreference the previous row's td in that index to see if it had a rowspan before generating the current row that way I could leave out the specific td, etc. There's a chance some javascript magic could be worked on that table element after it's generated but I've never really given it much thought.

**Reply by dmrickey** [reply]
What about something like this?

You could have an input option in the config that you can imply read and the configurator would have to know that they need to include that for cells that have been "spanned over". 

e.g. take a look at this one https://github.com/EddieDover/fvtt-party-sheet/pull/36/files. In row2 and three the last items are
```json
      {
        "name": "Spacer2",
        "type": "direct",
        "header": "hide",
        "text": ""
      }
      ...
      {
        "name": "Spacer9",
        "type": "direct",
        "header": "hide",
        "text": ""
      }
```
These are the cells directly under the single vision cell in the picture above. If those had something like `"inRowSpan: true"` then your hbs could do something like `{{#unless inRowSpan}} ...build the cell... {{/unless}}`

Or even more inline with it looks like your configuration already exists, they could be replaced with something like
```json
      {
        "type": "inRowSpan"
      }
      ...
      {
        "type": "inRowSpan"
      }
```

**Reply by EddieDover** [reply]
Yeah, this would work but we'll have to make it a dual-step process.

The user will need to:
- add `{"rowspan":X}` to their primary column
- add `"type":"span"` to their spacer columns

**Reply by EddieDover** [reply]
So in the example I've been using, if we made the Vision column a rowspan (and moved the content up into it)
```json
{
        "name": "Vision",
        "type": "direct-complex",
        "header": "show",
        "rowspan": 2,
        "text": [
          {
            "type": "exists",
            "value": "system.attributes.senses.darkvision",
            "text": "Darkvision: system.attributes.senses.darkvision"
          },
          {
            "type": "exists",
            "value": "system.attributes.senses.blindsight",
            "text": "Blindsight: system.attributes.senses.blindsight"
          },
          {
            "type": "exists",
            "value": "system.attributes.senses.tremorsense",
            "text": "Tremorsense: system.attributes.senses.tremorsense"
          },
          {
            "type": "exists",
            "value": "system.attributes.senses.truesight",
            "text": "Truesight: system.attributes.senses.truesight"
          },
          {
            "type": "exists",
            "value": "system.attributes.senses.special",
            "text": "Special: system.attributes.senses.special"
          }
        ]
      },
```

Then we added the new type to the spacer

```json
      {
        "name": "Ins",
        "type": "direct",
        "header": "show",
        "text": "system.skills.ins.passive"
      },
      {
        "name": "Spacer",
        "type": "span",  // <---- here
        "header": "hide",
        "text": ""
      },
      {
        "name": "Status",
        "type": "array-string-builder",
        "align": "left",
        "valign": "top",
        "maxwidth": 70,
        "header": "hide",
        "text": "statuses => value, "
      }
```

Then the output would look like:
![image](https://github.com/user-attachments/assets/29f16b8b-8cd4-4e79-b789-9a07a2c76931)

**Reply by EddieDover** [reply]
Making an Issue from this discussion. Let's continue discussing here, if necessary but you can pull the PR down and test yourself to make things easier.

**Reply by dmrickey** [reply]
Yeah that output looks great 👍. 

![image](https://github.com/user-attachments/assets/7cd11e03-16dc-4474-a10f-51670a777ea3)
(I haven't looked at your changes yet, but it seems like name/header/text wouldn't be needed if type is `span`)

**Reply by EddieDover** [reply]
>  (I haven't looked at your changes yet, but it seems like name/header/text wouldn't be needed if type is `span`)

Probably, but I'm not sure what will and won't break until I have more time to test, so I just left it in for now. After all these PRs are merged, we'll need to run through the documentation and provide examples / clean things up.
